### DISCUSSION TITLE
[A-L] (12 Gennaio 2022) Esercizio 1

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#237] [Esami A-L]
![image](https://user-images.githubusercontent.com/69389461/152745614-d38efc7f-3502-41ef-bb3d-82916ed26368.png)

**Comment by i0nut02** [comment]
1. 1 * 26^3 * 10^3 * 3
2. 26^3 * 10^3 * 3 + 26^4 * 10^3 - 26^3 * 10^3 
(password che iniziano con L + quelle che finiscono con ! - quelle che iniziano con L e finiscono con !)
3. 4 * 25^3 * 3 * 9^2 * 3
***
**EDIT (Matypist)**: contrassegnato come OffTopic perché non si capisce come mai 3 al posto di 4^2 (le possibili combinazioni dei caratteri speciali)

**Comment by neoSnakex34** [comment]
![ilgitubbo](https://user-images.githubusercontent.com/97003970/152766496-15fc3a91-0a3e-420d-a2cd-66f5ec2bdb25.jpg)

**Comment by AxnNxs** [comment]
> ![ilgitubbo](https://user-images.githubusercontent.com/97003970/152766496-15fc3a91-0a3e-420d-a2cd-66f5ec2bdb25.jpg)

Mi ritrovo con queste soluzioni.
I ragionamenti sono i seguenti:

1) Sono tutte le possibili combinazioni a meno della prima lettera fissata, per la quale avremo sempre una sola possibilità di scelta.

2) La domanda include una disgiunzione dei due casi, ma sappiamo che operando sugli stessi insiemi conteggeremo due volte le combinazioni incluse in entrambe. Preso per esempio il caso in cui fissiamo la prima lettera, conteremo involontariamente anche quelli che includono tutte le possibili combinazioni per l'ultimo simbolo, e lo stesso accadrà a parti invertite, quando vorremo fissare l'ultimo simbolo. Pertanto la soluzione sfrutta PIE ed è:

Da		AuB = A + B - AnB 
segue

(1 x 26 x 26 x 26 x 10 x 10 x 10 x 4 x 4) + (26 x 26 x 26 x 26 x 10 x 10 x 10 x 4 x 1) - (Tutte le combinazioni che contengono sia una "A" all'inizio che un "!" alla fine = 1 x 26 x 26 x 26 x 10 x 10 x 10 x 4 x 1)

3) A differenza del punto precedente, quando chiediamo le combinazioni possibili data una quantità fissata per un elemento, stiamo pensando in quante posizioni possiamo fissare quel dato. Pertanto avremo 4 possibilità di fissare A, poiché ve ne dovrà essere una sola, escludendo dal conteggio quella lettera per le altre posizioni, quindi 25 al posto di 26, e ragionamento analogo è da farsi per il numero 4  da esercizio. Il risultato è dato dal solito Principio Moltiplicativo.
